Feature/exposure materializations (#132)
dbt-labs/dbt-project-evaluator: README.md

<img width="500" alt="A refactored DAG showing three sources which are each being referenced by an accompanying staging model" src="https://user-images.githubusercontent.com/30663534/159603703-6e94b00b-07d1-4f47-89df-8e5685d9fcf0.png"> 

### Exposure Materializations
#### Model

> `fct_exposure_materializations` ([source](models/marts/dag/fct_exposure_materializations.sql)) shows each model with a direct relationship to an exposure that is *not* materialized as a table in the warehouse. 

Reviewer: I _do_ have opinions about the model name lol - I think this is a bit confusing because it makes me think it's related to the materialization of the actual exposure (which doesn't make sense). How about something like `fct_exposure_parents_materialization`?

Author: i'm cool with that!

Reviewer: Awesome! Looks like you already updated the model names but still need to update the readme to match!

Author: updated reference in README!

#### Reason to Flag

> Models that are referenced by an exposure are likely to be used heavily in downstream systems, and therefore need to be performant when queried. This model highlights instances where the models referenced by exposures are not either a `table` or `incremental` materialization.

Reviewer: I actually do think a screenshot of the DAG & example from our integration tests would be useful here, just to visualize that we're looking at the materialization of the model directly upstream of the exposure.

Author: agreed -- do you think i should try to somehow try to viusally capture tehir materializations? like renaming the nodes to be `view_` or `table_` prefixed?

Reviewer: That could work - or just add a note below the screenshot to explain the materialization of each model. The latter might be easier.

Author: added screenshot and note

- if a view is used downstream my *many* models, change materialization to table
- if a view has more complex calculations (window functions, joins between *many* tables, etc.), change materialization to table

> ### Exposure Materializations

Reviewer: Do you need to add a link to this section in the table of contents?

Reviewer: Also do we want to name this Exposure Parents Materializations to match the model name?

Author: we should do both!

#### Model

> `fct_exposure_parentes_materializations` ([source](models/marts/performance/fct_exposure_parentes_materializations.sql)) shows each model with a direct relationship to an exposure that is *not* materialized as a table in the warehouse. 

Reviewer: ```suggestion
`fct_exposure_parents_materializations` ([source](models/marts/performance/fct_exposure_parents_materializations.sql)) shows each model that is a direct parent of an exposure and is *not* materialized as a table in the warehouse. 
```

Author: i actually wanted this one to be in spanish

#### Example 

> <img width="500" alt="An example exposure with a table parent (fct_model_6) and an ephemeral parent (dim_model_7)" src="https://user-images.githubusercontent.com/73915542/178068955-742e2c87-4385-48f9-b9fb-94a1cbc8079a.png">

Reviewer: I would maybe choose a simpler example of something like `view_1` -> `exposure_1`. But ¯\_(ツ)_/¯

In this case, the parents of `exposure_1` are not both materialized as tables -- `dim_model_7` is ephemeral, while `fct_model_6` is a table. This model would return a record for the `dim_model_7 --> exposure_1` relationship. 

#### Reason to Flag

> Models that are referenced by an exposure are likely to be used heavily in downstream systems, and therefore need to be performant when queried. This model highlights instances where the models referenced by exposures are not either a `table` or `incremental` materialization.

Reviewer: Is there something we could reference here that explains why tables are more performant than views?

#### How to Remediate

> If necessary, update the `materialized` configuration on the models returned in `fct_exposure_materializations` to either `table` or `incremental`. This can be done in individual model files using a config block, or for groups of models in your `dbt_project.yml` file. See the docs on [model configurations](https://docs.getdbt.com/reference/model-configs) for more info!

Reviewer: ```suggestion
If necessary, update the `materialized` configuration on the models returned in `fct_exposure_parents_materializations` to either `table` or `incremental`. This can be done in individual model files using a config block, or for groups of models in your `dbt_project.yml` file. See the docs on [model configurations](https://docs.getdbt.com/reference/model-configs) for more info!
```
